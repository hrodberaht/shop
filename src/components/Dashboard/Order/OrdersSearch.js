import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class OrdersSearch extends Component {
  static propTypes = {
    handleCheck: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
  };

  render() {
    const { handleCheck, title } = this.props;
    return (
      <div>
        {title}
        <input type="checkbox" onClick={handleCheck} />
      </div>
    );
  }
}
