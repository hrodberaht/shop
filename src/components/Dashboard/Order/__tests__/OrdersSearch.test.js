import OrdersSearch from '../OrdersSearch';

const defaultProps = {
  handleCheck: jest.fn(),
  title: 'Search',
};

const setup = buildSetup(OrdersSearch, defaultProps);

describe('<OrdersSearch>', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleCheck if checkbox is clicked', () => {
    const { wrapper } = setup();
    wrapper.find('input').simulate('click');
    expect(wrapper.instance().props.handleCheck).toHaveBeenCalled();
  });
});
