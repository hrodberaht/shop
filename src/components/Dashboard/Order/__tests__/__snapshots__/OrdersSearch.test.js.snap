// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrdersSearch> should render without crash 1`] = `
<div>
  Search
  <input
    onClick={[MockFunction]}
    type="checkbox"
  />
</div>
`;
